### ISSUE TITLE
Unstoppable type inference (via Ctrl-C)

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#31142]
```
foo(x, ys...) = foo(x, ys)
foo(1, ())
```

yields 

```
^CInternal error: encountered unexpected error in runtime:
InterruptException()
jl_mutex_unlock at /Users/osx/buildbot/slave/package_osx64/build/src/./locks.h:138 [inlined]
jl_typeinf_end at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:2610
typeinf_ext at ./compiler/typeinfer.jl:577
typeinf_ext at ./compiler/typeinfer.jl:613
jfptr_typeinf_ext_1.clone_1 at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
jl_apply_generic at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:2219 [inlined]
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
jl_type_infer at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:277
jl_compile_method_internal at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1819
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1863
foo at ./REPL[4]:1
unknown function (ip: 0x12330c4e3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c413)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c343)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c273)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c1a3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c0d3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c003)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bf33)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330be63)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bd93)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bcc3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bbf3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bb23)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330ba53)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b983)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b8b3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b7e3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b713)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b643)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:323
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:362 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:759
jl_interpret_toplevel_thunk_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:885
Interpreter frame (ip: 1)
Core.CodeInfo(code=Array{Any, (3,)}[
  Expr(:call, Core.tuple),
  Expr(:call, :foo, 1, SSAValue(1)),
  Expr(:return, SSAValue(2))], codelocs=Array{Int32, (3,)}[1, 1, 1], method_for_inference_limit_heuristics=nothing, ssavaluetypes=3, linetable=Array{Any, (1,)}[Core.LineInfoNode(mod=Main, method=Symbol("top-level scope"), file=:none, line=0, inlined_at=0)], ssaflags=Array{UInt8, (0,)}[], slotflags=Array{UInt8, (0,)}[], slotnames=Array{Any, (0,)}[], inferred=false, inlineable=false, propagate_inbounds=false, pure=false)jl_interpret_toplevel_thunk at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:894
jl_toplevel_eval_flex at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:764
jl_toplevel_eval at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:773 [inlined]
jl_toplevel_eval_in at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:793
eval at ./boot.jl:328
eval_user_input at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:85
run_backend at /Users/yurivish/.julia/packages/Revise/yp5KG/src/Revise.jl:769
#58 at ./task.jl:259
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
start_task at /Users/osx/buildbot/slave/package_osx64/build/src/task.c:572
^V^C^C^C^C^C^C^CWARNING: Force throwing a SIGINT
Internal error: encountered unexpected error in runtime:
InterruptException()
is_derived_type at ./compiler/typelimits.jl:32
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:68
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type at ./compiler/typelimits.jl:57
is_derived_type_from_any at ./compiler/typelimits.jl:77
_limit_type_size at ./compiler/typelimits.jl:93
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
_limit_type_size at ./compiler/typelimits.jl:155
limit_type_size at ./compiler/typelimits.jl:21
jfptr_limit_type_size_5685.clone_1 at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
abstract_call_method at ./compiler/abstractinterpretation.jl:300
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:85
abstract_call at ./compiler/abstractinterpretation.jl:776
abstract_eval_call at ./compiler/abstractinterpretation.jl:805
abstract_eval at ./compiler/abstractinterpretation.jl:890
typeinf_local at ./compiler/abstractinterpretation.jl:1135
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1191
typeinf at ./compiler/typeinfer.jl:14
typeinf_edge at ./compiler/typeinfer.jl:497
abstract_call_method at ./compiler/abstractinterpretation.jl:345
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:85
abstract_call at ./compiler/abstractinterpretation.jl:776
abstract_eval_call at ./compiler/abstractinterpretation.jl:805
abstract_eval at ./compiler/abstractinterpretation.jl:890
typeinf_local at ./compiler/abstractinterpretation.jl:1135
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1191
typeinf at ./compiler/typeinfer.jl:14
typeinf_ext at ./compiler/typeinfer.jl:576
typeinf_ext at ./compiler/typeinfer.jl:613
jfptr_typeinf_ext_1.clone_1 at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
jl_apply_generic at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:2219 [inlined]
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
jl_type_infer at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:277
jl_compile_method_internal at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1819
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1863
foo at ./REPL[4]:1
unknown function (ip: 0x12330c683)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c5b3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c4e3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c413)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c343)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c273)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c1a3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c0d3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330c003)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bf33)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330be63)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bd93)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bcc3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bbf3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330bb23)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330ba53)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b983)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b8b3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b7e3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b713)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[4]:1
unknown function (ip: 0x12330b643)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:323
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:362 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:759
jl_interpret_toplevel_thunk_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:885
Interpreter frame (ip: 1)
Core.CodeInfo(code=Array{Any, (3,)}[
  Expr(:call, Core.tuple),
  Expr(:call, :foo, 1, SSAValue(1)),
  Expr(:return, SSAValue(2))], codelocs=Array{Int32, (3,)}[1, 1, 1], method_for_inference_limit_heuristics=nothing, ssavaluetypes=3, linetable=Array{Any, (1,)}[Core.LineInfoNode(mod=Main, method=Symbol("top-level scope"), file=:none, line=0, inlined_at=0)], ssaflags=Array{UInt8, (0,)}[], slotflags=Array{UInt8, (0,)}[], slotnames=Array{Any, (0,)}[], inferred=false, inlineable=false, propagate_inbounds=false, pure=false)jl_interpret_toplevel_thunk at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:894
jl_toplevel_eval_flex at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:764
jl_toplevel_eval at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:773 [inlined]
jl_toplevel_eval_in at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:793
eval at ./boot.jl:328
eval_user_input at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:85
run_backend at /Users/yurivish/.julia/packages/Revise/yp5KG/src/Revise.jl:769
#58 at ./task.jl:259
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
start_task at /Users/osx/buildbot/slave/package_osx64/build/src/task.c:572
^C^C^C^C^C^C^C^C^Cc⏎                                                                    $ cd /Users/yurivish/Dropbox/Projects/D4M\ Exploration ^C
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.0 (2019-01-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> foo(x, ys...) = foo(x, ys)
foo (generic function with 1 method)

julia> ^C

julia> foo(x, ys...) = foo(x, ys)
foo (generic function with 1 method)

julia> foo(1, ())
^CInternal error: encountered unexpected error in runtime:
InterruptException()
jl_mutex_unlock at /Users/osx/buildbot/slave/package_osx64/build/src/./locks.h:138 [inlined]
jl_typeinf_end at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:2610
typeinf_ext at ./compiler/typeinfer.jl:577
typeinf_ext at ./compiler/typeinfer.jl:613
jfptr_typeinf_ext_1.clone_1 at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
jl_apply_generic at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:2219 [inlined]
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
jl_type_infer at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:277
jl_compile_method_internal at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1819
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1863
foo at ./REPL[1]:1
unknown function (ip: 0x12621ed43)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621ec73)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621eba3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621ead3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621ea03)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e933)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e863)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e793)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e6c3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e5f3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e523)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e453)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e383)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e2b3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e1e3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e113)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621e043)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621df73)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621dea3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
foo at ./REPL[1]:1
unknown function (ip: 0x12621ddd3)
jl_fptr_trampoline at /Users/osx/buildbot/slave/package_osx64/build/src/gf.c:1864
do_call at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:323
eval_stmt_value at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:362 [inlined]
eval_body at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:759
jl_interpret_toplevel_thunk_callback at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:885
Interpreter frame (ip: 1)
Core.CodeInfo(code=Array{Any, (3,)}[
  Expr(:call, Core.tuple),
  Expr(:call, :foo, 1, SSAValue(1)),
  Expr(:return, SSAValue(2))], codelocs=Array{Int32, (3,)}[1, 1, 1], method_for_inference_limit_heuristics=nothing, ssavaluetypes=3, linetable=Array{Any, (1,)}[Core.LineInfoNode(mod=Main, method=Symbol("top-level scope"), file=:none, line=0, inlined_at=0)], ssaflags=Array{UInt8, (0,)}[], slotflags=Array{UInt8, (0,)}[], slotnames=Array{Any, (0,)}[], inferred=false, inlineable=false, propagate_inbounds=false, pure=false)jl_interpret_toplevel_thunk at /Users/osx/buildbot/slave/package_osx64/build/src/interpreter.c:894
jl_toplevel_eval_flex at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:764
jl_toplevel_eval at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:773 [inlined]
jl_toplevel_eval_in at /Users/osx/buildbot/slave/package_osx64/build/src/toplevel.c:793
eval at ./boot.jl:328
eval_user_input at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:85
macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:117 [inlined]
#26 at ./task.jl:259
jl_apply at /Users/osx/buildbot/slave/package_osx64/build/src/./julia.h:1571 [inlined]
start_task at /Users/osx/buildbot/slave/package_osx64/build/src/task.c:572
```
on Julia 1.1. The Julia process has to be killed; Ctrl-C does not succeed.

This is a reduced form of a bug that manifested somewhat differently (with a seemingly endless type inference stacktrace), but I think this might be at the core of it.

This code pattern came up when I had two definitions of `foo`, one with vararg `ys` and the other accepting an `NTuple` of a specific length, with the intention being that the vararg method calls the other one. (So this code was a bug on my part, but also a bug on the Julia side.)

**Comment by vtjnash** [comment]
If we annotate this, we can see that the runtime is cranking away at this as intended:
```julia
julia> f(x, a...) = (println(a); f(x, a))
f (generic function with 1 method)

julia> @code_typed f(1, ())
CodeInfo(
    @ REPL[1]:1 within `f'
1 ─     invoke Main.println(_3::Tuple{Tuple{}})::Any
│       invoke Main.f(_2::Int64, _3::Tuple{Tuple{}})::Union{}
└──     $(Expr(:unreachable))::Union{}
) => Union{}

julia> f(1, ())
((),)
(((),),)
((((),),),)
(((((),),),),)
((((((),),),),),)
(((((((),),),),),),)
((((((((),),),),),),),)
(((((((((),),),),),),),),)
((((((((((),),),),),),),),),)
...
```

However, a lot of the internal code (codegen, inference, subtyping, etc.) makes heavy use of try/catch/end to hide other internal issues—and we're accidentally also consuming SIGINT along the way.

**Comment by yurivish** [comment]
This seems fixed in Julia 1.6:

```julia
julia> foo(x, ys...) = foo(x, ys)
foo (generic function with 1 method)

julia> foo(1, ())
^CERROR: InterruptException:
Stacktrace:
 [1] foo(x::Int64, ys::Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{Tuple{}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}) (repeats 216 times)
   @ Main ./REPL[10]:1
 [2] top-level scope
   @ REPL[11]:1
```
